drivers/i2c/i2c_ll_stm32.c

```diff
@@ -292,6 +292,11 @@ static void i2c_stm32_irq_config_func_##name(struct device *dev)	\
 #define STM32_I2C_INIT(name)						\
 STM32_I2C_IRQ_HANDLER_DECL(name);					\
 									\
+COND_CODE_1(CONFIG_I2C_STM32_V2,					\
```

Comment:
We're trying to get limit the use of Kconfig flags and replace by compatibles whenever it is possible,
so here it would be:
```suggestion
COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(i2c_ll_stm32_v2),					\
```
However, I'm not sure this will work well, specially in the header file.
So if working, this would be nice, but this is not a blocking comment.

Reply:
Just tested this, and it doesn't work : it does compile, but the condition seems to be evaluated to false

Reply:
ok, we should find a way to do this better then.
Let's keep it as is, txs for the attempt